central/apitoken/expiration/expiration.go

```diff
@@ -0,0 +1,186 @@
+package expiration
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/apitoken/datastore"
+	"github.com/stackrox/rox/central/role/resources"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+)
+
+const (
+	notificationInterval = 1 * time.Hour      // 1 hour
+	staleNotificationAge = 24 * time.Hour     // 1 day
+	expirationWindow     = 7 * 24 * time.Hour // 1 week
+	expirationSlice      = 24 * time.Hour     // 1 day
+
+	// The timestamp format / layout is borrowed from `pkg/search/postgres/query/time_query.go`. It would be worth exporting.
+	timestampLayout = "01/02/2006 3:04:05 PM MST"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	expirySearchCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Integration),
+		),
+	)
+)
+
+// Notifier is the interface for a background task that notifies about API token expiration
+type Notifier interface {
+	Start()
+	Stop()
+}
+
+// ExpiringItemNotifier is the interface to list the tokens about to expire, and to send notifications
+// for items about to expire.
+type ExpiringItemNotifier interface {
+	ListItemsAboutToExpire() ([]search.Result, error)
+	Notify(identifiersToNotify []string) error
+}
+
+type expirationNotifierImpl struct {
```

Comment:
I think we should also inject a "clock" so we will be able to test it with mocked time.